src/channeld/automata/accept.rs

```diff
@@ -105,17 +110,71 @@ impl ChannelAccept {
                 "Accepted".ended(),
                 channel_id.actor(),
             ),
-            _ => todo!(),
+            ChannelAccept::Signed => {
+                format!("{} channel {:#} from a remote peer", "Signed".ended(), channel_id.actor(),)
+            }
+            ChannelAccept::Funded => {
+                format!("{} channel {:#} from a remote peer", "Funded".ended(), channel_id.actor(),)
+            }
+            ChannelAccept::Locked => {
+                format!("{} channel {:#} from a remote peer", "Locked".ended(), channel_id.actor(),)
+            }
         }
     }
 }
 
-fn finish_accepted(_event: Event<BusMsg>, _runtime: &mut Runtime) -> Result<ChannelAccept, Error> {
-    todo!()
+fn finish_accepted(event: Event<BusMsg>, runtime: &mut Runtime) -> Result<ChannelAccept, Error> {
+    let accept_event = match event.message {
+        BusMsg::Ctl(CtlMsg::Keyset(_, keys)) => {
+            runtime.state.channel.constructor_mut().set_local_keys(keys);
+
+            let accept_channel = runtime.state.channel.compose_accept_channel()?;
+            let accept_channel = LnMsg::AcceptChannel(accept_channel);
+
+            runtime.send_p2p(event.endpoints, accept_channel)?;
+            Ok(ChannelAccept::Signed)
+        }
+        wrong_msg => {
+            return Err(Error::UnexpectedMessage(wrong_msg, Lifecycle::Accepted, event.source))
+        }
+    };
+    accept_event
 }
 
-fn finish_signed(_event: Event<BusMsg>, _runtime: &mut Runtime) -> Result<ChannelAccept, Error> {
-    todo!()
+fn finish_signed(event: Event<BusMsg>, runtime: &mut Runtime) -> Result<ChannelAccept, Error> {
+    let signed_event = match event.message {
+        BusMsg::Bolt(LnMsg::FundingCreated(funding)) => {
+            let old_id = runtime
+                .state
+                .channel
+                .temp_channel_id()
+                .expect("Error to retrieve temporary channel id");
+            let channel_id = ChannelId::with(funding.funding_txid, funding.funding_output_index);
+            debug!("Changing channel id from {} to {}", runtime.identity(), channel_id);
+            runtime.set_identity(event.endpoints, channel_id).expect("unrecoverable ZMQ failure");
+            runtime.state.channel.update_from_peer(&LnMsg::FundingCreated(funding.clone()))?;
+            let new_id = runtime.state.channel.channel_id().expect("Error to retrieve channel id");
```

Comment:
Not sure why do we expect that the channel id is already set here

Reply:
Overall this change in the channel id from temporarily one to a permanent is a mess. We are forever sticked with the need to `expect` on both option values (temporary and permanent). I do not like it.

It seems that we need to separate an opened channel from an established channel the way it is done in c-lightning. I.e. their state machines should be different.

This is not related to this PR specifically, just a food for thoughts.